### ISSUE TITLE
Does not fill in the RuleId and Level fields of the SARIF result object

### ISSUE THREAD

**Issue by bdovaz** [mgechev/revive#1263]
Example:

https://github.com/oxsecurity/megalinter/actions/runs/13708328243/job/38338966390?pr=4837#step:7:1735

![Image](https://github.com/user-attachments/assets/52e4dd7b-71ec-492c-b13f-8026b3ebe874)

Although apparently it does fill them in:

https://github.com/mgechev/revive/blob/39ae68321ae71bb406f479ce9bdc72a19a12b298/formatter/sarif.go#L91

In the case of the `Level` field you can also choose to fill `runs -> tool -> driver -> rules -> rule -> defaultConfiguration -> level`. More information:

https://github.com/microsoft/sarif-tutorials/blob/main/docs/Authoring-rule-metadata-and-result-messages.md#sarif-representation

**Comment by alexandear** [comment]
@bdovaz Thank you for the issue. I analyzed the link you provided and found that you are trying to run `revive` on [this file](https://github.com/oxsecurity/megalinter/blob/0dfa730a9907ae2d7163555db13be78047d62396/.automation/test/golang/bad/golang_bad_02.go#L1):

```go
if len(in) == 0 {
  return "", fmt.Errorf("Input is empty")
}
```

`golang_bad_02.go` cannot compile and is not a valid Go file:

```sh
$ go build .automation/test/golang/bad/golang_bad_01.go
.automation/test/golang/bad/golang_bad_01.go:1:1: expected 'package', found 'if'
```

As a result, `revive` cannot determine the correct `ruleName` or set the proper `level` based on the `ruleName`.

I propose modifying `golang_bad_01.go` slightly by adding a `package` declaration and placing the `if` statement to a function. This way, `revive` can analyze it:

```go
package bad

func _() {
	if len(in) == 0 {
		return "", fmt.Errorf("Input is empty")
	}
}
```

After this change, you should be able to run `revive` successfully:

```sh
$ revive .automation/test/golang/bad/golang_bad_01.go
.automation/test/golang/bad/golang_bad_01.go:1:1: should have a package comment

```

---

@chavacava I think it's not a bug. We can close the issue as "Won't fix" or provide a workaround.
